File: be/src/main/java/com/example/mingle/domain/reservation/reservation/entity/Reservation.java
package com.example.mingle.domain.reservation.reservation.entity;

import com.example.mingle.domain.reservation.room.entity.Room;
import com.example.mingle.domain.user.user.entity.User;
import com.example.mingle.global.jpa.BaseEntity;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Entity
@Getter@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class Reservation extends BaseEntity {

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "room_id")
    private Room room;

    @Column(nullable=false)
    private LocalDate date;

    @Column(name = "start_time", nullable = false)
    private LocalTime startTime;

    @Column(name = "end_time", nullable = false)
    private LocalTime endTime;

    @Column(name = "reservation_status", nullable = false, length = 30)
    @Enumerated(EnumType.STRING)
    private ReservationStatus reservationStatus;

    @Column(length = 100)
    private String title;

}

File: fe/src/features/reservation/types/Reservation.ts
// features/reservation/types/reservation.ts

import { ReservationStatus } from "./reservationStatus";

export interface Reservation {
  id: number;
  roomId: number;
  roomName: string;
  roomType: "MEETING_ROOM" | "PRACTICE_ROOM";
  date: string; // yyyy-MM-dd
  startTime: string; // HH:mm
  endTime: string; // HH:mm
  reservationStatus: ReservationStatus;
  userId: number;
  userName: string;
  title: string | null; // 예약 제목
}

// 예약 폼 입력 데이터 타입
export interface ReservationFormInput {
  roomId: number;
  date: string;
  startTime: string;
  endTime: string;
  title: string;
  reservations?: Reservation[]; // 충돌 체크용 기존 예약 목록
}
